pos_payment_change/tests/test_module.py

```diff
@@ -0,0 +1,191 @@
+        order = self._sale(self.cash_payment_method, 35, self.bank_payment_method, 65)
+
+        order_qty = len(self.PosOrder.search([]))
+
+        with self.assertRaises(UserError):
+            # Should not work if total is not correct
+            self._change_payment(
+                order, self.cash_payment_method, 10, self.cash_payment_method, 10
+            )
+
+        self._change_payment(
+            order, self.cash_payment_method, 10, self.bank_payment_method, 90
+        )
+
+        self.bank_payment = self.session.order_ids.mapped("payment_ids").filtered(
+            lambda x: x.payment_method_id == self.bank_payment_method
+        )
+        self.cash_payment = self.session.order_ids.mapped("payment_ids").filtered(
+            lambda x: x.payment_method_id == self.cash_payment_method
+        )
+        # check Session
+        self.assertEqual(
+            self.cash_payment.amount,
+            10,
+            "Bad recompute of the balance for the statement cash",
+        )
+
+        self.assertEqual(
+            self.bank_payment.amount,
+            90,
+            "Bad recompute of the balance for the statement check",
+        )
+
+        # Check Order quantity
+        self.assertEqual(
+            order_qty,
+            len(self.PosOrder.search([])),
+            "In 'Update' mode, changing payment should not create" " other PoS Orders",
+        )
+
+    def test_02_payment_change_policy_refund(self):
+        self.pos_config.payment_change_policy = "refund"
+
+        # Make a sale with 35 in cash journal and 65 in check
+        order = self._sale(self.cash_payment_method, 35, self.bank_payment_method, 65)
+
+        order_qty = len(self.PosOrder.search([]))
+
+        self._change_payment(
+            order, self.cash_payment_method, 50, self.bank_payment_method, 50
+        )
+
+        # Check Order quantity
+        self.assertEqual(
+            order_qty + 2,
+            len(self.PosOrder.search([])),
+            "In 'Refund' mode, changing payment should generate" " two new PoS Orders",
+        )
+
+    def test_03_payment_change_closed_orders(self):
+        self.pos_config.payment_change_policy = "update"
+
+        # Make a sale with 35 in cash journal and 65 in check
+        order = self._sale(self.cash_payment_method, 35, self.bank_payment_method, 65)
+
+        self.session.state = "closed"
+
+        with self.assertRaises(UserError):
+            self._change_payment(
+                order, self.cash_payment_method, 10, self.bank_payment_method, 90
+            )
+
+    def test_04_payment_change_security(self):
+        self.pos_config.payment_change_policy = "refund"
+
+        order = self._sale(self.cash_payment_method, 35, self.bank_payment_method, 65)
+
+        # the demo user should be able to do this
+        user_demo = self.env.ref("base.user_demo")
+        wizard = (
+            self.PosPaymentChangeWizard.with_user(user_demo)
```

Comment:
Hello, @legalsylvain!
Tests are failing because Marc Demo stopped being related to pos manager group.
I don't want to spend time finding it out, why it happens.

How about just creating user with pos manager rights?